### PR TITLE
Color scale component

### PR DESCRIPTION
**Pull request by creilly8**
## Description
Improved color scale component code. 

Changes: 
1. Color scale configurable to show number line above or below the color bar (see unit test)
2. New unit tests
3. Replaced the color scale code in `client/src/block.tk.hicstraw.ts` with component
4. Added `markedValue` or a value that appears above the axis in the color bar for the hic app
5. Cleaned up import statements whilst searching for other places to use the new component. 

Test with: 
- Hic track: http://localhost:3000/?appcard=hic
- Hic app: http://localhost:3000/?genome=hg19&hicfile=proteinpaint_demo/hg19/hic/hic_demo.hic
- Unit tests: http://localhost:3000/testrun.html?dir=dom&name=colorScale.unit

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/dom/ColorScale.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	/** Optional. Size of the ticks in px. Default is 1. */
+	tickSize: number
+	/** Optional. Font size in px of the text labels. */
+	fontSize: number
+	/** Optional. Shows a value in the color bar for the default, bottom axis
+	 * This value cannot be zero at initialization.
+	 */
+	markedValue?: number | null
+
+	constructor(opts: {
```
Comment (xzhou82) [inline client/dom/ColorScale.ts:64]: constructor argument should be document at the beginning

File: client/dom/ColorScale.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
+import { scaleLinear } from 'd3-scale'
+import { Elem, SvgG } from '../types/d3'
```
Comment (xzhou82) [inline client/dom/ColorScale.ts:2]: wrong import. d3.d.ts

File: client/dom/ColorScale.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+	/** Optional. Font size in px of the text labels. */
+	fontSize: number
+	/** Optional. Shows a value in the color bar for the default, bottom axis
+	 * This value cannot be zero at initialization.
+	 */
+	markedValue?: number | null
+
+	constructor(opts: {
+		/** Required */
+		holder: Elem
```
Comment (xzhou82) [inline client/dom/ColorScale.ts:66]: wow didn't know can embed svg inside svg. works great!

File: client/src/block.mds2.ld.js [hunk 4]
```diff
@@ -1,4 +1,3 @@
-import * as common from '#shared/common'
```
Comment (xzhou82) [inline client/src/block.mds2.ld.js:1]: no need to update mds2, no way to test it and should have deleted these code

**Comment by xzhou82** [comment]
hic app has issue again. clicking on chrpair view to go into detailed view, click `Zoom In` the first time the contact canvas does not update. also `<<Detailed view` button shows up and shouldn't
@akanksha-r please test hic features

**Comment by xzhou82** [comment]
adding Alex and Airen as the reusable color scale should be used for disco (cnv) and scatter legends, so you can watch and assess

File: client/dom/ColorScale.ts (hunk 2 above)
Comment (creilly8) [reply client/dom/ColorScale.ts:2]: @xzhou82 `.d.ts` is the extension. We shouldn't need to include the extension on dictionary type files in a typescript file.

**Comment by creilly8** [comment]
> hic app has issue again. clicking on chrpair view to go into detailed view, click `Zoom In` the first time the contact canvas does not update. also `<<Detailed view` button shows up and shouldn't @akanksha-r please test hic features

@xzhou82 I'm not seeing this same issue. Would you share the URL or file you used to find this bug?

**Comment by airenzp** [comment]
Screenshot 2024-06-10 at 9 50 56 AM

This single tick with a one at the beginning is expected? I would think is better at the middle, if there is only one tick.

**Comment by creilly8** [comment]
> Screenshot 2024-06-10 at 9 50 56 AM
> This single tick with a one at the beginning is expected? I would think is better at the middle, if there is only one tick. Also better to put it oriented towards the bottom, as the min and max values.

@airenzp, Yes that's expected. It's the minimum cutoff, proportional to the overall scale. It's not a requirement to implement the scale, just how it's used for the Hi-C app and track. The unit tests show different ways to implement the scale.

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**